sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClient.java

```diff
@@ -0,0 +1,227 @@
+     * for encryption. This operation requires the keys/encrypt permission.
+     *
+     * <p>The {@link EncryptionAlgorithm encryption algorithm} indicates the type of algorithm to use for decrypting the specified encrypted content. Possible values
+     * for assymetric keys include: {@link EncryptionAlgorithm#RSA1_5 RSA1_5}, {@link EncryptionAlgorithm#RSA_OAEP RSA_OAEP} and {@link EncryptionAlgorithm#RSA_OAEP_256 RSA_OAEP_256}.
+     * Possible values for symmetric keys include: {@link EncryptionAlgorithm#A128CBC A128CBC}, {@link EncryptionAlgorithm#A128CBC_HS256 A128CBC-HS256},
+     * {@link EncryptionAlgorithm#A192CBC A192CBC}, {@link EncryptionAlgorithm#A192CBC_HS384 A192CBC-HS384}, {@link EncryptionAlgorithm#A256CBC A256CBC} and {@link EncryptionAlgorithm#A256CBC_HS512 A256CBC-HS512} </p>
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encrypted.
+     * @throws ResourceNotFoundException if the key cannot be found for encryption.
+     * @return The {@link EncryptResult} whose {@link EncryptResult#cipherText() cipher text} contains the encrypted content.
+     */
+    public EncryptResult encrypt(EncryptionAlgorithm algorithm, byte[] plaintext) {
+        return withContext(context -> client.encrypt(algorithm, plaintext, context, null, null)).block();
+    }
+
+    /**
+     * Decrypts a single block of encrypted data using the configured key and specified algorithm. Note that only a single block of data may be
+     * decrypted, the size of this block is dependent on the target key and the algorithm to be used. The decrypt operation
+     * is supported for both asymmetric and symmetric keys. This operation requires the keys/decrypt permission.
+     *
+     * <p>The {@link EncryptionAlgorithm encryption algorithm} indicates the type of algorithm to use for decrypting the specified encrypted content. Possible values
+     * for assymetric keys include: {@link EncryptionAlgorithm#RSA1_5 RSA1_5}, {@link EncryptionAlgorithm#RSA_OAEP RSA_OAEP} and {@link EncryptionAlgorithm#RSA_OAEP_256 RSA_OAEP_256}.
+     * Possible values for symmetric keys include: {@link EncryptionAlgorithm#A128CBC A128CBC}, {@link EncryptionAlgorithm#A128CBC_HS256 A128CBC-HS256},
+     * {@link EncryptionAlgorithm#A192CBC A192CBC}, {@link EncryptionAlgorithm#A192CBC_HS384 A192CBC-HS384}, {@link EncryptionAlgorithm#A256CBC A256CBC} and {@link EncryptionAlgorithm#A256CBC_HS512 A256CBC-HS512} </p>
+     *
+     * @param algorithm The algorithm to be used for decryption.
+     * @param cipherText The content to be decrypted.
+     * @throws ResourceNotFoundException if the key cannot be found for decryption.
+     * @return The decrypted blob.
+     */
+    public byte[] decrypt(EncryptionAlgorithm algorithm, byte[] cipherText, byte[] iv, byte[] authenticationData, byte[] authenticationTag) {
```

Comment:
>byte[] [](start = 11, length = 6)

We're updating all methods to return a data structure rather than a simple type for forwards compat.  This should return DecryptResult which has one property plaintext.

Reply:
added